task160_FinalTask/src/test/java/automationpractice/com/pages/BasePage.java

```diff
@@ -0,0 +1,57 @@
+package automationpractice.com.pages;
+
+import WebDriverSingleton.WebDriverSingleton;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.pagefactory.AjaxElementLocatorFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class BasePage {
+    private static final int TIMEOUT = 5;
+    private static final int POLLING = 100;
+    private WebDriverWait wait;
+    private JavascriptExecutor jse = (JavascriptExecutor) WebDriverSingleton.getInstance().getDriver();
```

Comment:
Move this part in WebDriverSignelton class as separate method

Reply:
done

I've created 3 separate methods in WebDriverSIngleton class:
getWebDriverWait
getJavascriptExecutor
getActions